### PR TITLE
Adding fortigate Cluster state check

### PR DESCRIPTION
**Pull request by epacke**
We discovered that one of our firewall nodes in a cluster had failed but it was not detected by LibreNMS.
This PR aims to fix this issue by using the following logic:

1. Is the monitored device clustered in any way?
2. If so, make sure the device count in `fgHaStatsEntry` is not 1

Also adding node state (master, backup and standalone) together with config sync state to ensure that the cluster config has been synced properly. 

*Credits to Elias in discord for the last checks*

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by epacke** [comment]
I am getting inconsistent results between the actual execution of the code and when debugging it.

```
SELECT 
    devices.device_id,
    status,
    status_reason,
    sensor_current,
    sensor_alert,
    lastupdate,
    sensor_prev,
    state_name,
    state_descr,
    state_value,
    state_generic_value,
    state_lastupdated
FROM 
    devices,
    sensors,
    sensors_to_state_indexes,
    state_indexes,
    state_translations
WHERE (
    state_name = "clusterState" AND
    devices.device_id = sensors.device_id AND
    sensors.sensor_id = sensors_to_state_indexes.sensor_id AND
    sensors_to_state_indexes.state_index_id = state_indexes.state_index_id AND
    state_indexes.state_index_id = state_translations.state_index_id
);
``` 

image

But `dd($clusterState);` returns 1

```
php ./discovery.php -h 92
...
#### Load disco module sensors ####
 ENTITY-SENSOR: Caching OIDs: entPhysicalDescr entPhysicalName entPhySensorType entPhySensorScale entPhySensorPrecision entPhySensorValue entPhySensorOperStatus
Airflow:
Current:
Charge:
Dbm:
Fanspeed:
Frequency:
Humidity:
Load:
Loss:
Power:
Power_consumed:
Power_factor:
Runtime:
Signal:
State: ^ 1
```

When running this in my docker container it works perfectly agains the same device though!

Both are running version 1.69.

**Comment by epacke** [comment]
I tried to remove and re-add one of my devices and then the check works fine. Any ideas of why that is the case?

**Comment by epacke** [comment]
RTFM. Discovery and sensor_current is only for the initial data. The rest would have to be done with a custom poller.

File: includes/definitions/discovery/fortigate.yaml [hunk 1]
```diff
@@ -6,68 +6,77 @@ modules:
         hardware: ENTITY-MIB::entPhysicalModelName.1
     processors:
         data:
-            -
-                oid: fgSysCpuUsage
-                num_oid: '.1.3.6.1.4.1.12356.101.4.1.3.{{ $index }}'
-                type: fortigate-fixed
+            - oid: fgSysCpuUsage
+              num_oid: ".1.3.6.1.4.1.12356.101.4.1.3.{{ $index }}"
+              type: fortigate-fixed
```
Comment (PipoCanaja) [inline includes/definitions/discovery/fortigate.yaml:11]: Hi @epacke 
Please avoid changing the structure of the YAML files. We have most of them following this structure.
Thanx

**Comment by murrant** [comment]
@epacke you are trying to work against the design of the system.  It is designed to handle tables just as the Fortigate MIB is written.

You just use the faHaStatsTable  (the proper name for faHaStatsEntry) and fgVdTable and pull out fgVdEntHaState and fgHaStatsSyncStatus.   If you want to exclude them sometimes, use the skip_values option, but is there a reason to exclude them?

It goes through each entry in the table and creates a sensor.  You do not want manually create sensors for each entry.

File: includes/definitions/discovery/fortigate.yaml (hunk 1 above)
Comment (epacke) [reply includes/definitions/discovery/fortigate.yaml:11]: Sorry, that was intentional

**Comment by epacke** [comment]
Sorry, I'll close this PR and will try to do some more homework. It's not ready for testing and I don't want to waste peoples time.

**Comment by murrant** [comment]
@epacke FYI, this was almost ready to be merged.

**Pull request closed without merging**